### PR TITLE
Deepgram streaming

### PR DESCRIPTION
**Pull request by matthewkennedy5**
## **User description**
attempting to be backwards compatible with whisper. 

when ASR_METHOD == "deepgram", deepgram keeps track of the transcription as it goes in `self.transcription`, which is used in start_response(). 

Next, I think deepgram has VAD functionality, so we can get rid of silero vad and call start_response() once deepgram has a transcript ready.

___

## **Description**
- Integrated Deepgram's live transcription service into the `ResponseAgent` class, replacing the previous static transcription setup.
- Added a new `transcript` attribute to `ResponseAgent` to store the ongoing transcription result from Deepgram.
- Set up event handling for live transcription results, updating the `transcript` attribute accordingly.
- Modified the `receive_audio` method to send audio data to Deepgram's live transcription service.
- Updated the `start_response` method to use the live transcription result when Deepgram is selected as the ASR method.
- Changed the default recording behavior in the `connect_daily` function to not record by default.
- Changed the TTS provider from ElevenLabs to local in the `connect_daily` function.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>response_agent.py</strong><dd><code>Integrate Deepgram Live Transcription</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

openduck-py/openduck_py/response_agent.py
<li>Replaced <code>PrerecordedOptions</code> and <code>FileSource</code> with <br><code>LiveTranscriptionEvents</code> and <code>LiveOptions</code> for DeepgramClient.<br> <li> Initialized DeepgramClient outside of the conditional check for <br><code>ASR_METHOD</code>.<br> <li> Added live transcription setup with event handling for Deepgram.<br> <li> Removed commented-out initialization of SileroVad.<br> <li> Added a new <code>transcript</code> attribute to store ongoing transcription.<br> <li> Updated <code>receive_audio</code> method to send audio data to Deepgram live <br>transcription.<br> <li> Modified <code>start_response</code> to use the stored <code>transcript</code> when <code>ASR_METHOD</code> <br>is set to "deepgram".<br> <li> Added <code>on_message</code> method to handle live transcription events and update <br>the <code>transcript</code>.

</details>
    

  </td>
  <td><a href="https://github.com/uberduck-ai/openduck/pull/101/files#diff-2a998d0b8029e572af7e5336e8b76fc6a8e411c591a2cb95768cf08dc6bb6f03">+41/-13</a>&nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>voice.py</strong><dd><code>Update Voice Router Configuration Defaults</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

openduck-py/openduck_py/routers/voice.py
<li>Changed the default value of <code>record</code> from <code>True</code> to <code>False</code> in <br><code>connect_daily</code> function.<br> <li> Updated the <code>TTSConfig</code> provider from "elevenlabs" to "local" in <br><code>connect_daily</code> function.

</details>
    

  </td>
  <td><a href="https://github.com/uberduck-ai/openduck/pull/101/files#diff-83ea5947a09005723a4450b92f794118b9e55099439455d9d9b4152c3ff0c4ce">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table><details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: openduck-py/openduck_py/response_agent.py [hunk 1]
```diff
@@ -254,6 +251,25 @@ def __init__(
...
+            interim_results=True,
+            utterance_end_ms="1000",
+            vad_events=True,
+        )
+
+        self.dg_connection.on(
+            LiveTranscriptionEvents.Transcript,
+            lambda x, result, **kwargs: self.on_message(result),
+        )
+        self.dg_connection.start(options)
```
Comment (zachwe) [inline openduck-py/openduck_py/response_agent.py:271]: Let's put this inside `if ASR_METHOD == "deepgram":`

File: openduck-py/openduck_py/response_agent.py [hunk 2]
```diff
@@ -483,3 +504,9 @@ async def speak_response(
             audio=np.frombuffer(audio_chunk_bytes, dtype=np.int16),
             latency=t_styletts - t_normalize,
         )
+
+    def on_message(self, result):
```
Comment (zachwe) [inline openduck-py/openduck_py/response_agent.py:508]: Call it `on_deepgram_message` or `on_streaming_asr_message`?

File: openduck-py/openduck_py/routers/voice.py [hunk 3]
```diff
@@ -279,7 +279,7 @@ async def connect_daily(
         session_id=session_id,
         record=record,
         input_audio_format="int16",
-        tts_config=TTSConfig(provider="elevenlabs", voice_id=voice_id),
+        tts_config=TTSConfig(provider="local", voice_id=voice_id),
```
Comment (zachwe) [inline openduck-py/openduck_py/routers/voice.py:282]: can we revert the changes in this file before merging? in prod it's nice to have recording and 11 labs voice for now

Comment (matthewkennedy5) [reply openduck-py/openduck_py/routers/voice.py:282]: ya

File: openduck-py/openduck_py/response_agent.py [hunk 4]
```diff
@@ -271,6 +288,9 @@ async def interrupt(self, task: asyncio.Task):
         self.is_responding = False
 
     async def receive_audio(self, message: bytes):
+
+        self.dg_connection.send(message)
```
Comment (zachwe) [inline openduck-py/openduck_py/response_agent.py:292]: Same thing here, should gate on `ASR_METHOD == "deepgram"`

**Review by zachwe — APPROVED** [review]
